src/libpmemfile-posix/block_array.c

```diff
@@ -112,9 +180,36 @@ is_zeroed(const void *addr, size_t len)
 	return true;
 }
 
+/*
+ * acquire_new_entry
+ * Allocates a new slot for block metadata inside a block_array associated
+ * with the file. This is not allocation using pmemobj, but merely finding
```

Comment:
Hm, so it can be misunderstood. "This" refers to allocation, i.e.: the act of allocating. What routine does.